### PR TITLE
Implement `Vec<String>`

### PR DESCRIPTION
**Pull request by Desdaemon**
Solves #191.

~~Blocker: Dart generator is creating unused code, I'm trying to find a workaround before declaring this mergeable.~~ Done!

### CODE REVIEW

**Comment by fzyzcjy** [comment]
> Blocker: Dart generator is creating unused code, I'm trying to find a workaround before declaring this mergeable. Done!

Congrats!

File: frb_codegen/src/generator_dart.rs [hunk 1]
```diff
@@ -361,7 +361,7 @@ fn generate_wire2api_func(ty: &ApiType, api_file: &ApiFile) -> String {
             }
         },
         Optional(opt) => format!(
-            "return raw == null ? null : _wire2api_{}(raw);",
+            "if (raw != null) return _wire2api_{}(raw);",
```
Comment (fzyzcjy) [inline frb_codegen/src/generator_dart.rs:363]: What about keeping the original code: it is explicit about what it returns and is thus more readable especially for new Dart users. Btw if you want to keep this change, maybe put it in a new PR, which will make the diff of PRs more readable.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -207,27 +207,25 @@ jobs:
         run: flutter test -d windows integration_test/main.dart --verbose
   flutter_linux_test:
     # CI on Linux does not work right now due to log reader errors
-    if: false
```
Comment (fzyzcjy) [inline .github/workflows/ci.yaml:210]: After CI is ok I will merge it

File: frb_example/pure_dart/dart/lib/main.dart [hunk 3]
```diff
@@ -97,6 +97,11 @@ void main(List<String> args) async {
       expect(listOfStructResp[3].width, 420);
     }
 
+    print('dart call handleStringList');
+    {
+      await api.handleStringList(names: ['Steve', 'Bob', 'Alex']);
```
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/lib/main.dart:102]: What about making a very simple assert to its return value, e.g. `expect(..., ...)`

File: frb_example/pure_dart/dart/pubspec.lock [hunk 4]
```diff
@@ -5,84 +5,84 @@ packages:
     dependency: transitive
     description:
       name: _fe_analyzer_shared
-      url: "https://pub.flutter-io.cn"
```
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/pubspec.lock:8]: Agree. I should have done that

**Comment by fzyzcjy** [comment]
Thank you for the PR! Only a few tiny suggestions and I am ready for merging ;)

File: frb_codegen/src/generator_dart.rs (hunk 1 above)
Comment (Desdaemon) [reply frb_codegen/src/generator_dart.rs:363]: Good point. I think I will revert this.

**Comment by fzyzcjy** [comment]
Just ping me when you are done and I will merge it

**Comment by Desdaemon** [comment]
I re-disabled Linux integration tests for now so please merge after this run @fzyzcjy

**Pull request merged**